### DISCUSSION TITLE
[psqldef] Add an ability to manage schema objects by schema restricted users.

### DISCUSSION THREAD

**Discussion by akisssa** [sqldef/sqldef#629] [Feature requests]
In the case of managing tables by schema restricted user, we cannot create table objects in other schema. This is because users can see table metadata outside of current schema with `pg_catalog.pg_namespace`. So I want to add an ability to psqldef.
   
There may be some options to implement this.

1. Skip if not match the current schema (https://github.com/sqldef/sqldef/blob/master/database/postgres/database.go#L98) in tableNames func. But this option may affect users who manage objects in multiple schemas. 
```go
		if schema != d.GetDefaultSchema() {
			continue
		}
```
2. Add an option like `--targetSchema` to psqldef and use it with the example 1 .

## Platform

* OS: Linux 
* RDBMS:  PostgreSQL 
* Version: v0.16.12

## --export output

None

## prepare 

```sql
CREATE SCHEMA scha;
CREATE USER usra WITH ENCRYPTED PASSWORD 'usra';
ALTER ROLE usra SET search_path TO scha;
GRANT ALL ON SCHEMA scha to usra;

CREATE SCHEMA schb;
CREATE USER usrb WITH ENCRYPTED PASSWORD 'usrb';
ALTER ROLE usrb SET search_path TO schb;
GRANT ALL ON SCHEMA schb to usrb;
```

## Input SQL

schema_a

```
CREATE table tbl_a (id int);
```

schema_b

```
CREATE table tbl_b (id int);
```

schema_b

## Current output

schema_a

```
-- Apply --
CREATE table tbl_a (id int);
```

schema_b
```
Error on DumpDDLs: sql: Scan error on column index 0, name "column_name": converting NULL to string is unsupported
exit status 1
```

## Expected output

schema_a

```
-- Apply --
CREATE table tbl_a (id int);
```

schema_b

```
-- Apply --
CREATE table tbl_b (id int);
```

**Comment by k0kubun** [comment]
(1) seems too specific to the one-schema-per-user model as you said. I'd prefer (2) since it seems like a generalization that can be achieved fairly easily.

As to the interface, WDYT about doing it over `--config` option? That configuration file already supports `target_tables`, so it seems more consistent to introduce something like `target_schema` there.

**Comment by akisssa** [comment]
@k0kubun Supporting `target_schema` option on `--config` looks better for this use case. I will try to fix it.

**Comment by akisssa** [comment]
Tried but it's scope is too large, so added a new `--targetSchema` option to the psqldef. Please check the pull request for details.
